index.js

```diff
@@ -84,10 +85,47 @@ Lexer.prototype = {
    * @api private
    */
 
-  tok: function(type, val){
-    return val === undefined ?
-      {type: type, line: this.lineno} :
-      {type: type, line: this.lineno, val: val};
+  tok: function(type, val, columnIncrement){
+    var res = {type: type, line: this.lineno};
+
+    if (this.colno !== null) res.col = this.colno;
+    if (val !== undefined) res.val = val;
+
+    if (columnIncrement !== undefined) this.incrementColumn(columnIncrement);
+
+    return res;
+  },
+
+  /**
+   * Increment `this.lineno` and reset `this.colno`.
+   *
+   * @param {Number} increment
+   * @api private
+  **/
+
+  incrementLine: function(increment, startingColumn){
+    if (increment !== undefined) {
+      this.lineno += increment;
+    } else {
+      ++this.lineno;
+    }
+
+    this.colno = startingColumn || 1;
```

Comment:
I feel like setting the `startingColumn` to anything other than `1` should be done via a separate, explicit call.